tests/src/python/test_provider_virtual.py

```diff
@@ -0,0 +1,657 @@
+# -*- coding: utf-8 -*-
+"""QGIS Unit tests for QgsVirtualLayerProvider
+
+.. note:: This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+"""
+__author__ = 'Hugo Mercier'
+__date__ = '26/11/2015'
+__copyright__ = 'Copyright 2015, The QGIS Project'
+# This will get replaced with a git SHA1 when you do a git archive
+__revision__ = '$Format:%H$'
+
+import qgis
+import os
+import tempfile
+import sys
+
+from qgis.core import (QGis,
+                       QgsVectorLayer,
+                       QgsFeature,
+                       QgsFeatureRequest,
+                       QgsField,
+                       QgsGeometry,
+                       QgsPoint,
+                       QgsMapLayerRegistry,
+                       QgsRectangle,
+                       QgsErrorMessage,
+                       QgsProviderRegistry,
+                       QgsVirtualLayerDefinition
+                       )
+
+from utilities import (unitTestDataPath,
+                       getQgisTestApp,
+                       TestCase,
+                       unittest
+                       )
+from providertestbase import ProviderTestCase
+from PyQt4.QtCore import *
+
+try:
+    from pyspatialite import dbapi2 as sqlite3
+except ImportError:
+    print "You should install pyspatialite to run the tests"
+    raise ImportError
+
+import tempfile
+
+# Convenience instances in case you may need them
+QGISAPP, CANVAS, IFACE, PARENT = getQgisTestApp()
+
+
+class TestQgsVirtualLayerProvider(TestCase):
```

Comment:
Can I suggest making this

    class TestQgsVirtualLayerProvider(TestCase, ProviderTestCase):

And adding the provider test data from `tests/testdata/provider` too? ProviderTestCase is designed to ensure consistent behaviour across all vector data providers, so will be good to identify any potential issues with this provider